tests/src/stronk_tests.cpp

```diff
@@ -424,4 +425,112 @@ TEST(can_hash, can_hash_overloads_std_hash)  // NOLINT
     }
 }
 
+struct a_size_string_type : stronk<a_size_string_type, std::string, can_size>
+{
+    using stronk::stronk;
+};
+
+struct a_size_vector_type : stronk<a_size_vector_type, std::vector<int>, can_size>
+{
+    using stronk::stronk;
+};
+
+TEST(can_size, can_size_works_for_strings)
+{
+    EXPECT_TRUE(a_size_string_type("").empty());
+    EXPECT_FALSE(a_size_string_type("a").empty());
+    for (size_t i = 0; i < 16; i++) {
+        EXPECT_EQ(a_size_string_type(std::to_string(i)).size(), std::to_string(i).size());
+        EXPECT_EQ(a_size_string_type(std::to_string(i)).empty(), std::to_string(i).empty());
+    }
+}
+
+TEST(can_size, can_size_works_for_vectors)
+{
+    for (size_t i = 0; i < 16; i++) {
+        EXPECT_EQ(a_size_vector_type(std::vector<int>(i)).size(), std::vector<int>(i).size());
+        EXPECT_EQ(a_size_vector_type(std::vector<int>(i)).empty(), std::vector<int>(i).empty());
+    }
+}
+
+struct a_can_iterate_vector_type : stronk<a_can_iterate_vector_type, std::vector<int>, can_iterate>
+{
+    using stronk::stronk;
+};
+
+TEST(can_const_iterate, can_const_iterate_works_for_vectors)
+{
+    for (size_t i = 0; i < 16; i++) {
+        const auto vector = [&i]()
```

Comment:
What determines that `vector` is of type `a_can_iterate_vector_type`?

Reply:
omg im noob